### DISCUSSION TITLE
Provide a section for path segments similar for query string params

### DISCUSSION THREAD

**Discussion by jjnilton** [Orange-OpenSource/hurl#3333] [Ideas] [answered]
The goal is to keep the request a bit more organized and easier to read/edit when working with long strings in the path segments.

So it could be something like:
```
POST http://example.com/{foo}/{bar}/example
Content-Type: application/json
[PathSegments]
foo: something
bar: something-else
```

**Comment by jcamiel** [comment, marked as answer]
Hi @jjnilton 

You can achieve something similar with `[Options]` section and variable:

```hurl
POST http://example.com/{{foo}}/{{bar}}/example
Content-Type: application/json
[Options]
variable: foo=something
variable: bar=something-else
```

**Reply by jjnilton** [reply]
That'll do, thanks!

**Reply by lambrospetrou** [reply]
@jcamiel Is there a strict order between evaluating all the `[Option]` of the request and when you evaluate/resolve the final URL to use?

For example, if I use the `skip` option to not execute a request, can we adapt Hurl so that the URL is not even evaluated?

I had a case where I would have `skip: true` since some variable was invalid, but because the URL was also attempting to evaluate it the test was failing, whereas if the URL was evaluated only after all the options are evaluated and the request will happen, then it wouldn't fail.

**Reply by jcamiel** [reply]
Hi @lambrospetrou 

I think we're evaluating options before anything else without doing anything special on `skip`. It seems OK to me that ˋskip` acts as a short circuit for everything. Don't know if it's easy to do but we can try to make it work!

**Reply by lambrospetrou** [reply]
I think it's fine for the options to be evaluated in unspecified order (or maybe as the user orders them if you want to), but at least if certain options like `skip` are enabled after evaluation, subsequent actions where the final resolved variables are used like the URL could be skipped.
